vizro-core/src/vizro/models/_controls/parameter.py

```diff
@@ -49,6 +47,16 @@ def check_target_present(cls, target):
             raise ValueError(f"Target {target_id} not found in model_manager.")
         return target
 
+    @validator("targets", each_item=True)
+    def check_data_frame_as_target_argument(cls, target):
+        targeted_argument = target.split(".", 1)[1]
+        if targeted_argument.startswith("data_frame") and targeted_argument.count(".") != 1:
+            raise ValueError(
+                f"Invalid target {target}. 'data_frame' target must be supplied in the form "
+                "<target_component>.data_frame.<dynamic_data_argument>"
+            )
+        return target
+
```

Comment:
btw @petar-qb I had go at adding the validation to see whether the targeted dataframe belongs to dynamic data and/or is a function that includes the right argument, and decided it's actually more effort than it's worth right now. We can add it some time in the future. Ideally it would work for "normal parameters that target non-dataframe also anyway.

Reply:
The bookmark is added in the https://github.com/McK-Internal/vizro-internal/issues/959 description.